src/main/host/thread_ptrace.c

```diff
@@ -341,7 +341,7 @@ static void _threadptrace_updateChildState(ThreadPtrace* thread, StopReason reas
             debug("child %d terminated by signal %d", thread->base.nativePid,
                   reason.exited_signal.signal);
             thread->childState = THREAD_PTRACE_CHILD_STATE_EXITED;
-            thread->returnCode = -1;
+            thread->returnCode = reason.exited_signal.signal + 128;
```

Comment:
Could you add a helper like `int return_code_for_signal(int signal)`? That'd consolidate the magic `+ 128` in one place, where we can also document that e.g. this is what bash does.

Reply:
Added. (And also now using this for preload return codes as well, previously it used a return code of -1.)